### ISSUE TITLE
Make sure the http module works with Python3

### ISSUE THREAD

**Issue by brifordwylie** [kbandla/dpkt#318]
Right now the http module isn't working with Python3. So I think the right thing to do with this is have it take bytes and then convert it to a unicode string and parse from there. 

@kbandla if you don't have the cycles for this, just assign to me.

**Comment by kbandla** [comment]
Looking into this

**Comment by kylekeppler** [comment]
@brifordwylie Can you provide any more detail? Unit tests pass under 3.5 for me. Perhaps a failing example?

**Comment by scott-hand** [comment]
@kylekeppler I'm not sure if it's what he was talking about but I'm currently having a miserable time trying to get dpkt.http to play nice with a gzipped HTTP response in a way that works with both Python 2 and 3. In Python 3, I have to do a `decode("latin-1")` on my string because StringIO demands it. When I unzip it later, I encode it back into latin-1 and everything is fine. In Python 2, StringIO breaks when it gets the data decoded to unicode because it tries to decode it as ascii.

The problem is that it's using StringIO but actually handling bytes data. I just tried changing my local copy of dpkt to use BytesIO in http.py and it fixed all of my problems.

I don't really have the familiarity with this library to test the impact of this with confidence, otherwise I'd make a one line diff and PR it. But I hope this helps.

**Comment by scott-hand** [comment]
Just to provide a bit more information, I was able to address my issue locally by replacing cStringIO with StringIO. The reason it has problems is noted in the documentation:

>Unlike the StringIO module, this module is not able to accept Unicode strings that cannot be encoded as plain ASCII strings.

The solution is either to dump cStringIO for portability (not a great option) or to rework http.py to use BytesIO. When I tried the latter on my local installation, it cleared up any issues Python 2 had with me passing a byte string in, but it needs some work in Python 3 to make everything place nice with using bytes.
